src/main/java/org/takes/http/BkBasic.java

```diff
@@ -116,4 +120,32 @@ private static Response failure(final Throwable err) {
         );
     }
 
+    /**
+     * Adds custom headers with information about socket.
+     * @param req Request
+     * @param socket Socket
+     * @return Request with custom headers
+     */
+    private static Request addSocketHeaders(final Request req,
+        final Socket socket) {
+        Request mreq = req;
+        mreq = new RqWithHeader(
```

Comment:
@ikhvostenkov I don't like this design. How about adding a method `withAdditionalHeader` to `RqWithHeader` that simply returns `new RqWithHeader(this, "Header-Name", "Header-Value")`. That way we can get a request with multiple headers through a fluent interface like this:

```java
return new RqWithHeader(mreq, "Header-1", "Value-1")
    .withAdditionalHeader("Header-2", "value-2")
    .withAdditionalHeader("Header-3", "value-3")
```